ledger/src/transaction_balances.rs

```diff
@@ -0,0 +1,68 @@
+use {
+    solana_account_decoder::{
+        parse_account_data::SplTokenAdditionalDataV2, parse_token::token_amount_to_ui_amount_v3,
+    },
+    solana_runtime::bank::TransactionBalancesSet,
+    solana_svm::transaction_balances::{BalanceCollector, SvmTokenInfo},
+    solana_transaction_status::{
+        token_balances::TransactionTokenBalancesSet, TransactionTokenBalance,
+    },
+};
+
+// decompose the contents of BalanceCollector into the two structs required by TransactionStatusSender
+pub fn compile_collected_balances(
+    balance_collector: Option<BalanceCollector>,
+) -> (TransactionBalancesSet, TransactionTokenBalancesSet) {
```

Comment:
do we ever need to call this if it is not `Some`? we should have a `BalanceCollector` if we have `TransactionStatusSender`, right?

So I think we may not need to handle the case when `balance_collector` is `None`.

Reply:
we do have to handle the `None` case if transaction processing blows the program cache limit https://github.com/anza-xyz/agave/pull/5588#discussion_r2045340500

i think im going to return `Option<_>` here and fill the vec with vecs tho to be closer to existing behavior

Reply:
i did something slightly different but same effect, we produce vectors ful of empty vectors, rather than mere empty vectors

Reply:
(resolving as outdated)